src/Validation/Validation.php

```diff
@@ -653,12 +660,12 @@ public static function decimal($check, $places = null, $regex = null)
      * Only uses getmxrr() checking for deep validation, or
      * any PHP version on a non-windows distribution
      *
-     * @param string $check Value to check
+     * @param mixed $check Value to check
      * @param bool $deep Perform a deeper validation (if true), by also checking availability of host
      * @param string|null $regex Regex to use (if none it will use built in regex)
      * @return bool Success
      */
-    public static function email($check, $deep = false, $regex = null)
+    public static function email($check, ?bool $deep = false, ?string $regex = null): bool
```

Comment:
I think the second argument is not nullable

Reply:
The method has check like `if ($deep === false || $deep === null) {` plus there are tests which pass `null` for second param.

Reply:
tests are invalid :)